mono/tests/Makefile.am

```diff
@@ -1078,14 +1122,26 @@ TESTSAOT_IL=$(TESTS_IL:.exe=.exe$(PLATFORM_AOT_SUFFIX))
 TESTSAOT_BENCH=$(TESTS_BENCH:.exe=.exe$(PLATFORM_AOT_SUFFIX))
 TESTSAOT_STRESS=$(TESTS_STRESS:.exe=.exe$(PLATFORM_AOT_SUFFIX))
 TESTSAOT_GSHARED=$(TESTS_GSHARED:.exe=.exe$(PLATFORM_AOT_SUFFIX))
+#FIXME
+#TESTSAOT_TAILCALL=$(TEST_TAILCALL:.exe=.exe$(PLATFORM_AOT_SUFFIX))
 endif
 
 EXTRA_DIST=test-driver test-runner.cs $(TESTS_CS_SRC) $(TESTS_IL_SRC) \
-	$(TESTS_BENCH_SRC) $(TESTS_STRESS_SRC) stress-runner.pl
+	$(TESTS_BENCH_SRC) $(TESTS_STRESS_SRC) stress-runner.pl \
+	$(TEST_TAILCALL_CS_SRC) \
+	$(TEST_TAILCALL_IL_SRC)
 
 %.exe: %.il
+	-mkdir -p $(@D)
 	$(ILASM) -out:$@ $<
 
+# FIXME missing dependencies on BCL
+%.exe_test_tailcall_done: %.exe Makefile $(srcdir)/../../runtime/mono-wrapper ../mini/mono
```

Comment:
You want this target to be `test-tailcall` so you can run it multiple times without having to cleanup everytime.

Reply:
This is incrementally correct. Once something succeeds, it doesn't have to rerun unless the test or runtime changes.